evm/p2p/les.py

```diff
@@ -6,14 +18,159 @@
 
 class Status(Command):
     _cmd_id = 0
+    decode_strict = False
+    # A list of (key, value) pairs is all a Status msg contains, but since the values can be of
+    # any type, we need to use the raw sedes here and do the actual deserialization in
+    # decode_payload().
+    structure = sedes.CountableList(sedes.List([sedes.binary, sedes.raw]))
+    # The sedes used for each key in the list above.
+    items_sedes = {
+        'protocolVersion': sedes.big_endian_int,
+        'networkId': sedes.big_endian_int,
+        'headTd': sedes.big_endian_int,
+        'headHash': sedes.binary,
+        'headNum': sedes.big_endian_int,
+        'genesisHash': sedes.binary,
+        'serveHeaders': None,
+        'serveChainSince': sedes.big_endian_int,
+        'serveStateSince': sedes.big_endian_int,
+        'txRelay': None,
+        'flowControl/BL': sedes.big_endian_int,
+        'flowControl/MRC': sedes.CountableList(
+            sedes.List([sedes.big_endian_int, sedes.big_endian_int, sedes.big_endian_int])),
+        'flowControl/MRR': sedes.big_endian_int,
+    }
+
+    def decode_payload(self, rlp_data):
+        data = super(Status, self).decode_payload(rlp_data)
+        # The LES/Status msg contains an arbitrary list of (key, value) pairs, where values can
+        # have different types and unknown keys should be ignored for forward compatibility
+        # reasons, so here we need an extra pass to deserialize each of the key/value pairs we
+        # know about.
+        decoded = {}
```

Comment:
Instead of building up local state here, can we instead use the `eth_utils.to_dict` decorator and change this to use `yield key, value` to build the return data.